### DISCUSSION TITLE
Editing Attributions in Avatar admin panel breaks the whole avatar system

### DISCUSSION THREAD

**Discussion by blairmacintyre** [mozilla/hubs#4704] [Hubs Cloud Questions]
This is related to https://github.com/mozilla/hubs/discussions/3412

If I go to the avatars admin page, and hit edit, I will see something like this:
![image](https://user-images.githubusercontent.com/1348861/135722601-a83c38f1-e4cf-49c8-9919-abfe22244f7d.png)

Notice that attributions is showing as "[Object object]" which looks like how JS would print an array of one objects in the console.

I had entered "readyplayerme" when I uploaded the object, and back in the object list, Attributions shows as {"creator":"readyplayerme"}

I I edit that field and enter "{"creator":"readyplayerme"}" in text, things break.  I no longer see any avatars in the Avatar selection panel for the server, and if I go and look at the server logs I see:
```
Oct 01 17:34:33 confident-thief.realitymedia-internal.net bash[918]: reticulum.default@realitymedia-hub(O): ** (ArgumentError) cannot load "{\"creator\":\"readyplayer.me\"}" as type :map for field :attributions in %Ret.Avatar{bin_owned_file_id: nil, orm_map_owned_file_id: nil, imported_from_port: nil, account: #Ecto.Association.NotLoaded, emissive_map_owned_file: #Ecto.Association.NotLoaded, parent_avatar_id: nil, avatar_sid: nil, bin_owned_file: #Ecto.Association.NotLoaded, gltf_owned_file: #Ecto.Association.NotLoaded, description: nil, gltf_owned_file_id: nil, slug: nil, avatar_listings: #Ecto.Association.NotLoaded, state: nil, allow_remixing: nil, parent_avatar_listing: #Ecto.Association.NotLoaded, attributions: nil, base_map_owned_file_id: nil, account_id: nil, allow_promotion: nil, parent_avatar: #Ecto.Association.NotLoaded, emissive_map_owned_file_id: nil, imported_from_host: nil, inserted_at: nil, base_map_owned_file: #Ecto.Association.NotLoaded, parent_avatar_listing_id: nil, normal_map_owned_file: #Ecto.Association.NotLoaded, updated_at: nil, thumbnail_owned_file: #Ecto.Association.NotLoaded, reviewed_at: nil, meta: #Ecto.Schema.Metadata<:loaded, "ret0", "avatars">, orm_map_owned_file: #Ecto.Association.NotLoaded, imported_from_sid: nil, avatar_id: nil, thumbnail_owned_file_id: nil, name: nil, normal_map_owned_file_id: nil}
``` 

It doesn't matter what I enter, it's impossible to change that field back to anything that isn't borking the server.

I _suspect_ what is happening is that the things being received from the server is JSON, which is what is printed in the listing.  But, the edit panel is operating on the Object created from the JSON;  when a string is entered, and it's converted BACK to JSON, it's invalid (since it's an escaped string, not a valid JSON).

I would try and fix this but I don't really know React (!) (so I don't understand how the admin code is getting the data from the server, or how it's connected to the React panels), and when I run the client locally, any time I try to got to the admin panel is just shows me the homepage.

**Comment by pattersonbl2** [comment]
Local install normally won't connect to admin pages due to being unable to connect to the reticulum server.  I would have to look into the error code for both problems you have stated because it seems that your reticulum server is having issues accepting the request on your AWS setup.  Do you remember any modifications to the existing code base?

**Reply by blairmacintyre** [reply]
Not sure I understand your reply.   My local dev server connects to my reticulum server just fine.  I don't know what this means: "it seems that your reticulum server is having issues accepting the request on your AWS setup".   The hubs-cloud server and such are running just fine. 

Over on discord, @netpro2k  suggested that I need to run the admin server locally as well (in the hubs/admin directory), but I can't see how to configure that to work.  If you know how to run that locally, that would be great.   Specifically, I don't know what to set ITA_SERVER and POSTGREST_SERVER to in the .env file in the admin directory

**Comment by blairmacintyre** [comment]
Ok, I have a partial fix, courtesy of @netpro2k  (if I knew React, perhaps I could have figured this out):
Change
```
      <TextInput source="attribution"/>
```
to
``` 
      <TextInput source="attribution" format={JSON.stringify} parse={JSON.parse}/>
```
This "mostly works", enough for me to fix the issues with my avatar.  Now when I go look at the "attribute" field I edited to be `{"creator":"GT Avatar Maker"}` the field shows `"{\"creator\":\"GT Avatar Maker\"}"` .  Pasting the correct one on, and saving, fixes it.  Irritatingly, the wrong values shows up fine in the avatar list, probably because the escape characters are handled while displaying.

The remaining problem is that if the field is not valid JSON you can't edit;  your only option  is to paste valid json in.  So a "complete" fix would be to check if the field is valid; if it isn't valid, show an error and disallow saving.
